src/node_buffer.cc

```diff
@@ -51,12 +51,37 @@
 #define BUFFER_MALLOC(length)                                               \
   zero_fill_all_buffers ? calloc(length, 1) : malloc(length)
 
-#define SWAP_BYTES(arr, a, b)                                               \
-  do {                                                                      \
-    const uint8_t lo = arr[a];                                              \
-    arr[a] = arr[b];                                                        \
-    arr[b] = lo;                                                            \
-  } while (0)
+#if defined(__GNUC__) || defined(__clang__)
+#define BSWAP_INTRINSIC_2(x) __builtin_bswap16(x)
+#define BSWAP_INTRINSIC_4(x) __builtin_bswap32(x)
+#define BSWAP_INTRINSIC_8(x) __builtin_bswap64(x)
+#elif defined(__linux__)
+#include <byteswap.h>
+#define BSWAP_INTRINSIC_2(x) bswap_16(x)
+#define BSWAP_INTRINSIC_4(x) bswap_32(x)
+#define BSWAP_INTRINSIC_8(x) bswap_64(x)
+#elif defined(_MSC_VER)
+#include <intrin.h>
+#define BSWAP_INTRINSIC_2(x) _byteswap_ushort(x);
+#define BSWAP_INTRINSIC_4(x) _byteswap_ulong(x);
+#define BSWAP_INTRINSIC_8(x) _byteswap_uint64(x);
+#else
+#define BSWAP_INTRINSIC_2(x) (x << 8) | (x >> 8)
+#define BSWAP_INTRINSIC_4(x)                                                  \
+  ((x & 0xFF) << 24) |                                                        \
+  ((x & 0xFF00) << 8) |                                                       \
+  ((x >> 8) & 0xFF00) |                                                       \
+  ((x >> 24) & 0xFF)
+#define BSWAP_INTRINSIC_8(x)                                                  \
+  ((x & 0xFF00000000000000ull) >> 56) |                                       \
+  ((x & 0x00FF000000000000ull) >> 40) |                                       \
+  ((x & 0x0000FF0000000000ull) >> 24) |                                       \
+  ((x & 0x000000FF00000000ull) >> 8)  |                                       \
+  ((x & 0x00000000FF000000ull) << 8)  |                                       \
+  ((x & 0x0000000000FF0000ull) << 24) |                                       \
+  ((x & 0x000000000000FF00ull) << 40) |                                       \
+  ((x & 0x00000000000000FFull) << 56)
```

Comment:
Could you wrap the `x`es in the macros in parentheses, like `(x)`? I don’t think it makes a difference right now, it’s just a good practice to do that.